pom.xml

```diff
@@ -3342,6 +3342,24 @@
 
         <!-- Default compiler options - enable additional lint checks -->
         <plugins>
+            <plugin>
```

Comment:
Since this is hard to review fully (for example myself, @mandy-chessell , @davidradl  have all seen slightly different behaviour),I think we should try out changes just on a limited set of modules until we are comfortable with how they work. We can then be more free to experiment until we get a good solution (which this may be)

Reply:
It would be great if you could compile a list of strange behaviours in order to be able to address them. I think that even without this PR the Maven "Don't detect" setting mentioned in the first comment of the PR would solve the "duplicate class" issue starting with a clean project structure.

Reply:
My concern is that if we go with the current config wide-scale across the project (an approach I did initially support!) we may hit various errors as we're moving away from the default location for source as assumed by maven/plugins. I know that isn't precise, but last time round we had problems that we couldn't fully describe. Maybe with time. Egeria has so many modules, and the overall build process is non trivial.

I also don't know what else the 'don't detect' may change. Usually with intellij we would have auto-import on, and all of intellijs settings around maven etc come from importing the pom (rather than making specific intellij config), so what else does that setting do? Does it affect the import of other modules

As such I think a better initial step is to keep the scope narrow/limited. This seems to me to be through
 * limiting the projects (ie plugin at module level) we use lombok on until a little time/changes have passed (in master) so we are comfortable with them (we also have multiple non-java modules). That gives us more confidence before rolling out more widely
 * Trying to stick with defaults for 'regular' java code, and adding any moves/special steps only for lombok -- for example here I'm thinking about things like having lombok source files in a seperate directory (this seems common practice?) so that only they are processed via the plugin?

Reply:
having researched/tested/discussed further I now support the proposal in the PR including going back to a change across all projects. If any further issues arise an Issue should be raised. Other ideas all seem to have their own set of issues and are not as good.